modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -402,6 +417,28 @@ protected Experiment internalGetExperiment(Application.Name appName,
     }
 
     /**
+     * Updates the tags for a given Application.
+     *
+     * @param applicationName the Application for which the tags should be added
+     * @param tags            list of tags for the experiment
+     */
+    protected void updateExperimentTags(Application.Name applicationName, Set<String> tags) {
+
+        if (tags == null)
+            tags = Collections.EMPTY_SET;
+
+        // update the tag table, just rewrite the complete entry
+        List<ExperimentTagsByApplication> listAllTags = experimentAccessor.getAllTags(applicationName.toString()).all();
```

Comment:
@AndreaSuckro I do understand that it might happen in a rare scenario and I completely agree that it might not be super important to do it for this feature but because this is new code we are adding, I strongly feel that we do it the right way. It will help in making the API more performant too as you no longer would need to query the complete experiment table each time you update a tag. I understand that this is late in the cycle and would be a little time consuming, therefore I am OK with merging this change provided we have a card to address this in our prioritized backlog. But I would still suggest to do it now if you could spare some time addressing it. Your call! :)